subsys/net/ip/net_context.c

```diff
@@ -977,7 +977,6 @@ int net_context_accept(struct net_context *context,
 
 	if (net_context_get_ip_proto(context) == IPPROTO_TCP) {
 		ret = net_tcp_accept(context, cb, user_data);
-		goto unlock;
```

Comment:
I would not say this is useless. It tries to protect a case where someone puts code before the unlock: label and forgets to add the goto.